preset: Additional presets and revisions for 1.1.0.

bradleysepos opened the pull request:
See #159 and #201 for previous work.

### References

- Apple TV 4K
  - https://www.apple.com/apple-tv-4k/specs/
- Chromecast Ultra
  - https://developers.google.com/cast/docs/media
- Fire TV 4K
  - https://developer.amazon.com/docs/fire-tv/device-specifications.html
- Playstation (PS4 Pro)
  - http://manuals.playstation.net/document/en/ps4/music/mp_format_m.html
- Roku Ultra
  - https://www.roku.com/products/roku-ultra
  - https://support.roku.com/article/208754908-how-to-use-roku-media-player-to-play-your-videos-music-and-photos
  - https://support.roku.com/article/211194478-what-is-4k-ultra-hd-resolution-
- Vimeo
  - https://vimeo.com/help/compression
- Xbox
  - https://forums.plex.tv/discussion/294993/xbox-one-x-directplay-and-4k
- YouTube
  - https://support.google.com/youtube/answer/1722171

### Samples for testing

Sample files for testing are available at https://www.dropbox.com/sh/lqx4z0t6goyrzxk/AAClhQ-94FMIFYcRBHDVcxN1a?dl=0&lst=. Be sure to download the original files (not the web preview Dropbox creates) and test on the intended devices.

- Test on iPhone 7, 7 Plus, 8, 8 Plus, X, Apple TV 4K
  - ✅ BBBS Apple 2160p60 4K HEVC Surround.mp4
  - ✅ ToS Apple 2160p60 4K HEVC Surround.mp4
- Test on Chromecast Ultra
  - BBBS Chromecast 2160p60 4K HEVC Surround.mp4
  - ToS Chromecast 2160p60 4K HEVC Surround.mp4
- Test on Fire TV 2nd, 3rd Generation
  - BBBS Fire TV 2160p60 4K HEVC Surround.mp4
  - ToS Fire TV 2160p60 4K HEVC Surround.mp4
- ~~~Test on Roku Premiere, Premiere+~~~
  - ~~~BBBS Roku 2160p60 4K Surround.mkv~~~
  - ~~~ToS Roku 2160p60 4K Surround.mkv~~~

### Changes since HandBrake 1.0.x

#### General

- No changes. Encoding methods for HEVC and 4K are becoming more standardized, but more time is needed before HandBrake ships general presets for these.

#### Web (added in 86fbaa29e4c0527c522d4e7ed0250d6a5fa3a389)

- Add `Vimeo YouTube HQ 2160p60 4K`
  - *High quality H.264 video (up to 2160p60) and high bit rate AAC stereo audio in an MP4 container. Compatible with video hosting services supporting 4K video, such as Vimeo and YouTube.*
- Add `Vimeo YouTube HQ 1440p60 2.5K`
  - *High quality H.264 video (up to 1440p60) and high bit rate AAC stereo audio in an MP4 container. Compatible with video hosting services supporting 2.5K video, such as Vimeo and YouTube.*
- Add `Vimeo YouTube HQ 1080p60`
  - *High quality H.264 video (up to 1080p60) and high bit rate AAC stereo audio in an MP4 container. Compatible with video hosting services supporting 1080p60, such as Vimeo and YouTube.*
- Add `Vimeo YouTube HQ 720p60`
  - *High quality H.264 video (up to 720p60) and high bit rate AAC stereo audio in an MP4 container. Compatible with video hosting services supporting 720p60, such as Vimeo and YouTube.*
- Add `Vimeo YouTube 720p30`
  - *H.264 video (up to 720p30) and high bit rate AAC stereo audio in an MP4 container. Compatible with most video hosting services, such as Vimeo and YouTube.*

#### Devices

- Add `Apple 2160p60 4K HEVC Surround`
  - *H.265 video (up to 2160p60), AAC stereo audio, and Dolby Digital (AC-3) surround audio, in an MP4 container. Compatible with Apple iPhone 7, 7 Plus, 8, 8 Plus, X, and later; Apple TV 4K.*
- Add `Chromecast 2160p60 4K HEVC Surround`
  - *H.265 video (up to 2160p60), AAC stereo audio, and Dolby Digital (AC-3) surround audio, in an MP4 container. Compatible with Google Chromecast Ultra.*
- Add `Fire TV 2160p60 4K HEVC Surround`
  - *H.265 video (up to 2160p60), AAC stereo audio, and Dolby Digital (AC-3) audio, in an MP4 container. Compatible with Amazon Fire TV 2nd Generation and later.*
- Rename `Roku 2160p60 4K Surround` → `Roku 2160p60 4K HEVC Surround`
  - *H.265 video (up to 2160p60), AAC stereo audio, and surround audio, in an MKV container. Compatible with Roku 4, Streaming Stick+, Premiere+, and Ultra.*
  - Change is limited to preset name (to reflect HEVC) and mention more models in description
- ~~~Add `Roku 2160p60 4K Surround`~~~
  - ~~~*H.264 video (up to 2160p60), AAC stereo audio, and surround audio, in an MKV container. Compatible with Roku 4, Premiere, Premiere+, and Ultra.*~~~
- Revise descriptions to reflect compatible devices released since HandBrake 1.1.0

#### Matroska

- Add `H.265 MKV 2160p60`
  - *H.265 video (up to 2160p60) and AAC stereo audio, in an MKV container.*
- Add `H.264 MKV 2160p60`
  - *H.264 video (up to 2160p60) and AAC stereo audio, in an MKV container.*
- Add `VP9 MKV 2160p60`
  - *VP9 video (up to 2160p60) and Opus stereo audio, in an MKV container.*
- Revise order: H.265 moved above H.264 and VP9 moved above VP8

#### Production (added in 86fbaa29e4c0527c522d4e7ed0250d6a5fa3a389)

- Add `Production Max`
  - *Maximum bit rate, constant frame rate H.264 video and high bit rate AAC stereo audio in an MP4 container. For professional use as an intermediate format for video editing. Creates gigantic files.*
- Add `Production Standard`
  - *High bit rate, constant frame rate H.264 video and high bit rate AAC stereo audio in an MP4 container. For professional use as an intermediate format for video editing. Creates very large files.*
- Add `Production Proxy 1080p`
  - *Intra-only, constant frame rate H.264 video (up to 1080p) and high bit rate AAC stereo audio in an MP4 container. For professional use as a low resolution proxy format for video editing.*
- Add `Production Proxy 540p`
  - *Intra-only, constant frame rate H.264 video (up to 540p) and high bit rate AAC stereo audio in an MP4 container. For professional use as a low resolution proxy format for video editing.*

### TODO

- Android 4K?
- Playstation 4K - *Initial research suggests PS4 Pro support only and no HEVC.*
- Xbox 4K - *Initial research suggests 4K cannot be played from USB devices. Streaming and Blu-ray only.* 😼

bradleysepos committed:
preset: Additional presets and revisions for 1.1.0.

bradleysepos commented:
Added some 60 fps encodes of Big Buck Bunny Sunflower version for testing high frame rate.

Removed the Apple 4K preset that used H.264; Apple says it should work but apparently it doesn't, at least not everywhere intended. HEVC verified working on iPhone 7 Plus so we'll stick to just that preset for 4K on Apple devices.

bradleysepos commented:
Short update, it appears 2160p60 (or greater than 30 fps) requires sdtp atom which the mp4 muxer does not currently write. So the Big Buck Bunny Sunflower version encode will not work on Apple TV 4K and possibly others until we rectify this.

bradleysepos commented:
Included bd76fff to allow Apple TV 4K to support 2160p60 in MP4.

galad87 commented:
Could we use x265 10bit instead of the 8bit? I think most of the hevc hardware decoders supported 10bit since the beginning.

bradleysepos commented:
I discussed this briefly on #handbrake-dev. Scott raised the issue of possible confusion since the pipeline is 8-bit, which I think is valid. The 10-bit test files I made also came out larger. So maybe we’ll switch these to 10-bit once the rest of the pipeline is ready.

bradleysepos commented:
Samples at the Dropbox link have been updated with new encodes based on John's fixes.

bradleysepos commented:
I looked into 4K media support on current Playstation and Xbox consoles. Initial research suggests:

- PS4 Pro supports 4K media but HEVC is not supported; other Playstation consoles will not play 4K media at all (verified)
- All current Xbox models support 4K HEVC but only for streaming media (Plex, etc.) and Blu-ray; USB devices need not apply

As I only have an original PS4, I cannot test 4K on the PS4 Pro. If someone would like to test on their hardware, I'll consider making a Playstation 4K preset.

The Xbox situation seems unfortunate. The only benefit to a 4K preset would be to avoid transcoding when streaming via Plex or similar, and I cannot test.

bradleysepos commented:
Rebased. Minor description change to list more Roku models. Dropped non-HEVC Roku 4K; appears only the higher end models support it and they support HEVC too, so let's go with that officially.
